core-rust/state-manager/src/metrics.rs

```diff
@@ -473,3 +486,73 @@ impl SelfProposalMissTracker {
         self.gauge.add(new_missed_count - outdated_missed_count);
     }
 }
+
+/// A number of most recent [`ProposerTimestampDatapoint`]s to track for metrics purposes.
+const PROGRESS_RATE_HISTORY_LEN: usize = 10;
```

Comment:
How was `10` calculated?

Reply:
It is a wild guess :D
I am happy with any number that in practice translates to a "few seconds, up to 1 minute" order of magnitude.
(exposing an average-over-durations longer than Prometheus scraping interval [15-60 sec] does not make much sense)
I am open to more educated guesses.

Reply:
With current round rate (3/s - which should go up) 10 seems a little bit low. 20, 30?

Reply:
> no server-side throttling there?

yup, there's no throttling at the moment

170 is a lot, but plausible

Reply:
maybe 100 slots then? If we're syncing fast we don't care that the sample is for a shorter period, we're almost certainly healthy anyway (right?)

Reply:
Alright, I adjusted it to 100, lemme merge then 👍